### DISCUSSION TITLE
How to exclude object from export for JPG/PNG the right way?

### DISCUSSION THREAD

**Discussion by klavsbuss** [fabricjs/fabric.js#7578] [Q&A] [answered]
I'm trying to exclude particular object from export using `excludeFromExport: true`, but it's still visible if export to jpg, png. Removing object before export and then adding it back after export seems like wrong approach. What would be right way of doing this?

Example:
https://jsfiddle.net/4dfsLa18/1/

**Comment by asturur** [comment, marked as answer]
i think the simple way to handle this, is to set visible to false before exporting.
I don't remember a simpler way.

**Reply by klavsbuss** [reply]
Thanks, got it. In this case will change visibility upon export.

**Comment by asturur** [comment]
To give some context. This wasn't planned.
JSON (and SVG!) is considered an export. It was considered common to export for some undo/redo system to export very often an so an option was created. Also to add to canvas objects that are more UI and not part of the document structure.

While PNG/JPEG are a simple render.

I m not saying this is right, but as of now, you have to make object invisible if you don't want them to render ( or export ) in a jpeg/png
